pkg/controller/common/annotation/controller_version.go

```diff
@@ -65,7 +65,7 @@ func UpdateControllerVersion(ctx context.Context, client k8s.Client, obj runtime
 		return err
 	}
 	log.V(1).Info("updating controller version annotation", "namespace", namespace, "name", name, "kind", obj.GetObjectKind())
-	return client.Update(obj)
+	return errors.WithStack(client.Update(obj))
```

Comment:
Can't we call `WithStack()` at a higher level (ie. in the top-level Reconcile() function) to automatically add the stack for all errors?

This seems to be doing what I would expect:

```go

package main

import (
	"fmt"

	"github.com/pkg/errors"
)

func main() {
	err := myfuncWithStack()
	fmt.Printf("%+v", errors.WithStack(err))
}

func myfuncWithStack() error {
	return myfunc()
}

func myfunc() error {
	return returnerr()
}

func returnerr() error {
	return errors.New("function error")
}
```

```
❯ go run tmp.go
function error
main.returnerr
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/tmp.go:27
main.myfunc
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/tmp.go:23
main.myfuncWithStack
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/tmp.go:19
main.main
        /Users/seb/work/go/src/github.com/elastic/cloud-on-k8s/tmp.go:14
runtime.main
        /usr/local/Cellar/go/1.14.3/libexec/src/runtime/proc.go:203
```

Reply:
But that will only work for errors created with `github.com/pkg/errors`or similar libraries because they capture a stack trace in the call to `New`?

Reply:
Oh yes @pebrc you are right!